.github/workflows/go.yml

```diff
@@ -0,0 +1,155 @@
+name: Build and Test
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build-and-test:
+    name: Build Go and Test
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
```

Comment:
from a brief internet search, steps cannot run in parallel, only jobs. So by putting this all in one job, the windows, mac and linux builds are now sequential instead of parallel. This will slow down CI by quite a bit I'd imagine.

I think it will be faster to split these into separate jobs and also clearer as all the steps are mixing up the different OS steps.